gatsby-node.js

```diff
@@ -2,106 +2,92 @@ const path = require("path")
+  const firstPromise = await graphql(
+    `
+      {
+        allMarkdownRemark(
+          sort: { fields: [frontmatter___date], order: DESC }
+          limit: 1000
+        ) {
+          group(field: frontmatter___categories) {
+            fieldValue
+          }
+          edges {
+            node {
+              fields {
+                slug
               }
-              edges {
-                node {
-                  fields {
-                    slug
-                  }
-                  frontmatter {
-                    title
-                  }
-                }
+              frontmatter {
+                title
               }
             }
           }
-        `
-      ).then(result => {
-        if (result.errors) {
-          console.log(result.errors)
-          reject(result.errors)
         }
-
-        const categories = result.data.allMarkdownRemark.group
-        const posts = result.data.allMarkdownRemark.edges
-
-        categories.forEach(category => {
-          const path = `/Categories/${category.fieldValue}`
-          createPage({
-            path,
-            component: catePage,
-            context: {
-              category: category.fieldValue,
-            },
-          })
-        })
-
-        posts.forEach((post, index) => {
-          const previous =
-            index === posts.length - 1 ? null : posts[index + 1].node
-          const next = index === 0 ? null : posts[index - 1].node
-          const blogPath = `/blogs/${_.kebabCase(post.node.frontmatter.title)}`
-          createPage({
-            path: blogPath,
-            component: blogPost,
-            context: {
-              slug: post.node.fields.slug,
-              previous,
-              next,
-            },
-          })
-        })
-      })
-    )
-    resolve(
-      graphql(
-        `
-          {
-            allMarkdownRemark(limit: 1000) {
-              group(field: frontmatter___author) {
-                fieldValue
-              }
-            }
+      }
+    `
+  )
+  if (firstPromise.errors) {
+    reporter.panicOnBuild(`Error while running Graphql query.`)
+  }
+  const secondPromise = await graphql(
+    `
+      {
+        allMarkdownRemark(limit: 1000) {
+          group(field: frontmatter___author) {
+            fieldValue
           }
-        `
-      ).then(result => {
-        if (result.errors) {
-          console.log(result.errors)
-          reject(result.errors)
         }
+      }
+    `
+  )
+  if (secondPromise.errors) {
+    reporter.panicOnBuild(`Error while running Graphql query.`)
+  }
+  const blogPost = path.resolve("./src/templates/blog-post.tsx")
+  const catePage = path.resolve("./src/templates/category-template.tsx")
+  const authorPage = path.resolve("./src/templates/author-template.tsx")
+
+  const categories = firstPromise.data.allMarkdownRemark.group
+  const posts = firstPromise.data.allMarkdownRemark.edges
+  const authors = secondPromise.data.allMarkdownRemark.group
```

Comment:
It looks like you are doing two queries so you can use the graphql `group` field for both `categories` and `authors`. See this article about [graphql field aliases](https://graphqlmastery.com/blog/how-to-use-graphql-aliases).

Reply:
I think the code is outdated. Once I read the article about using 2 promises. I updated. After reading the "graphql field aliases", I realize I did not need to use 2 promises, and I update the query and just use one promise. I think the code is outdated. Right now, I am figuring out how to pass the author name from the index page to the author page. I tried to use `localStorage.setitem()` and `localStorage.getItem()`. The result is super weird. It keeps passing only one author name. This PR still needs time to work on.